### PR TITLE
YONK-703: User Progress list API

### PR DESCRIPTION
**Pull request by aamishbaloch**
User Progress list API to return progress of the user in enrolled courses.

### CODE REVIEW

**Comment by aamishbaloch** [comment]
@ziafazal Can you review this?

File: edx_solutions_api_integration/users/serializers.py [hunk 1]
```diff
@@ -128,3 +128,46 @@ class UserRolesSerializer(serializers.Serializer):
     """ Serializer for user roles """
     course_id = serializers.CharField()
     role = serializers.CharField()
+
+
+class CourseProgressSerializer(serializers.Serializer):
+    """ Serializer for course progress """
+    created = serializers.CharField()
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/serializers.py]: Should not this be a `DateTimeField`?

File: edx_solutions_api_integration/users/tests.py [hunk 2]
```diff
@@ -2353,3 +2354,203 @@ def test_user_courses_grades_list_get_after_enrollment(self):  # pylint: disable
...
+        )
+
+        cls.content_subchild = ItemFactory.create(
+            category="video",
+            parent_location=cls.content_child2.location,
+            data=cls.test_data,
+            display_name="Child Video",
+        )
+
+        cls.overview = ItemFactory.create(
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/tests.py]: We propably shoud have simple course structure to test user's progress use cases. No need to have about and other sections. Just a chapter, sequential, veritcal and couple of vidoes should be enough.

File: edx_solutions_api_integration/users/tests.py [hunk 3]
```diff
@@ -17,6 +17,8 @@
 from edx_notifications.lib.publisher import register_notification_type, publish_notification_to_user
 import mock
 import before_after
+from edx_solutions_api_integration.courses.content import TEST_STATIC_TAB2_CONTENT, TEST_STATIC_TAB1_CONTENT, \
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/tests.py]: No need for these. See the comment below.

File: edx_solutions_api_integration/users/tests.py [hunk 4]
```diff
@@ -2353,3 +2354,203 @@ def test_user_courses_grades_list_get_after_enrollment(self):  # pylint: disable
...
+            start=cls.course_start_date,
+            end=cls.course_end_date,
+            org="MTD"
+        )
+
+        cls.user = UserFactory()
+        cache.clear()
+
+    @classmethod
+    def add_course_updates(cls):
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/tests.py]: I think we don't need it for user progress test cases.

File: edx_solutions_api_integration/users/tests.py [hunk 5]
```diff
@@ -2353,3 +2354,203 @@ def test_user_courses_grades_list_get_after_enrollment(self):  # pylint: disable
...
+                 u'id': 1},
+            ],
+            u'data': u''}
+        return updates
+
+    def test_users_progress_list(self):
+        """ Test progress value returned by users progress list api """
+
+        # enroll user in the course
+        test_uri = '{}/{}/courses'.format(self.base_users_uri, self.user.id)
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/tests.py]: Nit: We should user `CourseEnrollmentFactory` here to improve speed of test.
```
CourseEnrollmentFactory.create(user=self.user, course_id=self.course.id)
```

File: edx_solutions_api_integration/users/views.py [hunk 6]
```diff
@@ -2,7 +2,11 @@
 
 import json
 import logging
+
+from course_metadata.models import CourseAggregatedMetaData
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/views.py]: Import order here does not comply with [edx strandards](http://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python_guidelines.html#imports-order).

File: edx_solutions_api_integration/users/views.py [hunk 7]
```diff
@@ -1554,3 +1560,31 @@ def post(self, request, user_id, msg_id):
...
+
+    def get(self, request, user_id):  # pylint: disable=unused-argument
+        try:
+            user = User.objects.get(id=user_id)
+        except ObjectDoesNotExist:
+            raise Http404
+        queryset = CourseEnrollment.objects.filter(user=user)
+
+        student_progress = StudentProgress.objects.filter(user=user).values('course_id', 'completions')
+        course_overview = CourseOverview.objects.all().values('id', 'start', 'end', 'course_image_url', 'display_name')
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/views.py]: I have concerns with pulling everything from `CourseOverview` and `CourseAggregatedMetaData` as iterating over large dataset would have performance impact.

**Comment by aamishbaloch** [comment]
Done with the changes you suggested. @ziafazal Can you review the changes?

File: edx_solutions_api_integration/users/views.py [hunk 8]
```diff
@@ -2,7 +2,10 @@
 
 import json
 import logging
+
 from edx_notifications.lib.consumer import mark_notification_read
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/views.py]: import order for these two lines is still not right.

File: edx_solutions_api_integration/users/views.py [hunk 9]
```diff
@@ -56,6 +60,8 @@
     HasOrgsFilterBackend
 from edx_solutions_api_integration.models import GroupProfile, APIUser as User
 from edx_solutions_organizations.serializers import BasicOrganizationSerializer
+
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/views.py]: nit: additional line.

**Pull request merged**